paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentSheetViewModelTest.kt

```diff
@@ -828,26 +834,74 @@ internal class PaymentSheetViewModelTest {
             )
     }
 
-    fun `Verify supported payment methods exclude afterpay if no shipping`() {
+    @Test
+    fun `Verify supported payment methods exclude afterpay if no shipping and no allow flag`() {
+        val viewModel = createViewModel(
+            args = ARGS_CUSTOMER_WITH_GOOGLEPAY.copy(
+                config = ARGS_CUSTOMER_WITH_GOOGLEPAY.config?.copy(
+                    shippingDetails = null,
+                    allowsPaymentMethodsRequiringShippingAddress = false,
+                )
+            )
+        )
+
         viewModel.setStripeIntent(
             PaymentIntentFixtures.PI_WITH_SHIPPING.copy(
                 paymentMethodTypes = listOf("afterpay_clearpay"),
-                shipping = null
+                shipping = null,
             )
         )
 
-        assertThat(viewModel.supportedPaymentMethods.size).isEqualTo(0)
+        assertThat(viewModel.supportedPaymentMethods).isEmpty()
+    }
+
+    @Test
+    fun `Verify supported payment methods include afterpay if allow flag but no shipping`() {
+        val viewModel = createViewModel(
+            args = ARGS_CUSTOMER_WITH_GOOGLEPAY.copy(
+                config = ARGS_CUSTOMER_WITH_GOOGLEPAY.config?.copy(
+                    shippingDetails = null,
+                    allowsPaymentMethodsRequiringShippingAddress = true,
+                )
+            )
+        )
 
         viewModel.setStripeIntent(
             PaymentIntentFixtures.PI_WITH_SHIPPING.copy(
-                paymentMethodTypes = listOf("afterpay_clearpay")
+                paymentMethodTypes = listOf("afterpay_clearpay"),
             )
         )
 
-        assertThat(viewModel.supportedPaymentMethods.size).isEqualTo(1)
-        assertThat(viewModel.supportedPaymentMethods.first()).isEqualTo(
-            lpmRepository.fromCode("afterpay_clearpay")!!
+        val expectedPaymentMethod = lpmRepository.fromCode("afterpay_clearpay")
+        assertThat(viewModel.supportedPaymentMethods).containsExactly(expectedPaymentMethod)
+    }
+
+    @Test
+    fun `Verify supported payment methods include afterpay if shipping but no allow flag`() {
+        val viewModel = createViewModel(
+            args = ARGS_CUSTOMER_WITH_GOOGLEPAY.copy(
+                config = ARGS_CUSTOMER_WITH_GOOGLEPAY.config?.copy(
+                    shippingDetails = AddressDetails(
+                        name = "Test Name",
+                        address = PaymentSheet.Address(
+                            line1 = "123 Main Street",
+                            postalCode = "12345",
+                            country = "US",
+                        ),
+                    ),
```

Comment:
I might be misreading this shippingDetails's presence or absence should have no effect on the test, right?

Reply:
Same confusion as above 🙈 Changed it now.